py/server/deephaven/experimental/sql.py

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+#
+import contextlib
+import inspect
+import jpy
+from typing import Dict, Union, Mapping, Optional, Any
+
+from deephaven import DHError
+from deephaven.table import Table, _j_py_script_session
+
+_JSql = jpy.get_type("io.deephaven.engine.sql.Sql")
+
+
+@contextlib.contextmanager
+def _scope(globals: Mapping[str, Any], locals: Mapping[str, Any]):
+    j_py_script_session = _j_py_script_session()
+    # Can do `globals | locals` in Python 3.9+
+    j_py_script_session.pushScope({**globals, **locals})
+    try:
+        yield
+    finally:
+        j_py_script_session.popScope()
+
+
+def eval(
```

Comment:
`eval` is a built-in python function, so we are better off renaming this to `sql_eval` or something else.

Reply:
I'm hoping we always reference it qualified by the sql module:

```python
from deephaven.experimental import sql

sql.eval(...)
```

Or, I could get behind the more explicit name `evaluate(...)`.

Reply:
The user can always do that themselves:
`from deephaven.experimental.sql import eval as sql_eval`

Reply:
Overwriting a built-in function is generally considered to be bad practice.

Reply:
I've changed the function name to "evaluate"